docs/reference-guides/block-api/block-bindings.md

```diff
@@ -0,0 +1,285 @@
+| -------- | ------- |
+| Paragraph  | content    |
+| Heading | content     |
+| Image    | ID, url, title, alt    |
+| Button    | text, url, linkTarget, rel    |
+
+## Registering a custom source
+
+Registering a source requires defining at least `name` and a `callback` function that gets a value from the source and passes it back to a block attribute.
+
+Once a source is registered, any supporting block's `metadata.bindings` attribute can be configured to read a value from that source.
+
+Registration can be done on the server via PHP or in the editor via JavaScript, and both can coexist.
+
+The label defined in server registration will be overridden by the label defined in the editor.
+
+### Server registration
+
+Server registration allows applying a callback that will be executed on the frontend for the defined bound attribute.
+
+The function to register a custom source is `register_block_bindings_source($name, $args)`.
+
+- `name`: `string` that sets the unique ID for the custom source.
+- `args`: `array` that contains:
+    - `label`: `string` with the human-readable name of the custom source.
+    - `uses_context`: `array` with the block context that is passed to the callback (optional).
+    - `get_value_callback`: `function` that will run on the bound block render function. It accepts three arguments: `source_args`, `block_instance` and `attribute_name`. This value can be overriden with the filter `block_bindings_source_value`.
+
+Note that `register_block_bindings_source()` should be called from a handler attached to the `init` hook.
+
+Here is an example:
+
+```php
+add_action(
+	'init',
+	function() {
+		register_block_bindings_source(
+			'wpmovies/visualization-date',
+			array(
+				'label'              => __( 'Visualization Date', 'custom-bindings' ),
+				'get_value_callback' => function( array $source_args, $block_instance ) {
+					$post_id = $block_instance->context['postId'];
+					$visualization_date = get_post_meta( $post_id, 'wp_movies_visualization_date', true );
+					if ( ! $visualization_date ) {
+						$visualization_date = date("m/d/Y");
+					}
+					return $visualization_date;
+				},
+				'uses_context'       => array( 'postId' ),
+			)
+		);
+	}
+);
+```
+
+This example needs a `post_meta` registered:
+
+```php
+add_action(
+	'init',
+	function() {
+		register_meta(
+			'post',
+			'wp_movies_visualization_date',
+			array(
+				'show_in_rest' => true,
+				'single'       => true,
+				'type'         => 'string',
+				'label'	       => __( 'Movie visualization date' ),
+			)
+		);
+	}
+);
+```
+
+#### Block bindings source value filter
+
+The value returned by `get_value_callback` can be modified with the `block_bindings_source_value` filter.
+The filter has the following parameters:
+
+- `value`: The value to be filtered.
+- `name`: The name of the source.
+- `source_args`: Array containing source arguments.
+- `block_instance`: The block instance object.
+- `attribute_name`: The name of the attribute.
+
```

Comment:
Actually, the example for the custom source `wpmovies/visualization-date` could be simplified to skip the condition for computing the default date. Instead, it could be presented as an example of using `block_bindings_source_value` filter.

Reply:
Fixed in https://github.com/WordPress/gutenberg/pull/66251/commits/a2e5a7ac2d1ff2eafbb1688ebf820bc9b20bc268